core/src/nucleus/ribosome/callback/receive.rs

```diff
@@ -1,17 +1,59 @@
-use super::call;
 use crate::{
     context::Context,
-    nucleus::ribosome::callback::{Callback, CallbackParams, CallbackResult},
+    nucleus::{
+        ribosome::{
+            self,
+            callback::{Callback, CallbackParams, CallbackResult},
+            Defn,
+        },
+        ZomeFnCall,
+    },
 };
+use holochain_core_types::{error::HolochainError, json::JsonString};
 use std::sync::Arc;
 
+#[derive(Serialize, Deserialize, Debug, DefaultJson)]
+struct ReceiveReturnValue(Result<String, String>);
+
 pub fn receive(
     context: Arc<Context>,
     zome: &str,
     // we ignore params for genesis
     params: &CallbackParams,
 ) -> CallbackResult {
-    call(context, zome, &Callback::Receive, params)
+    let params = match params {
+        CallbackParams::Receive(payload) => payload,
+        _ => return CallbackResult::NotImplemented,
+    };
+
+    let zome_call = ZomeFnCall::new(
+        zome,
+        "no capability since this is a callback",
```

Comment:
I thought we weren't calling this capability anymore? Trait?

Reply:
We haven't removed "capability" from the code yet. Otherwise this function wouldn't expect a capability name as second parameter.

Reply:
K